docs/linux/dm.md

```diff
@@ -319,82 +323,74 @@ bitmaps.
+used as a backend . This reduces the amount of metadata needed to manage valid
 blocks (blocks written). The on-disk metadata format is as follows:
 
-1. The first block of the first randomly writable zone found contains the
-super block which describes the amount and position on disk of metadata blocks. 
-
-2. Following the super block, a set of blocks is used to describe the mapping
-of the logical chunks of the target logical device to data zones. The mapping
-is indexed by logical chunk number and each mapping entry indicates the data
-zone storing the chunk data and optionally the zone number of a random zone
-used to buffer random modification to the chunk data.
-
-3. A set of blocks used to store bitmaps indicating the validity of blocks in
-the data zones follows the mapping table blocks. A valid block is a block that
-was writen and not discarded. For a buffered data zone, a block can be valid
-only in the data zone or in the buffer zone.
-
-To protect internal metadata against corruption in case of sudden power loss or
-system crash, two sets of metadata zones are used. One set, the primary set, is
-used as the main metadata set, while the secondary set is used as a log.
-Modified metadata are first written to the secondary set and the log so created
-validated by writing an updated super block in the secondary set. Once this log
-operation completes, updates in place of metadata blocks can be done in the
+1. **Super Block**: The first block of the first randomly-writable zone that is found contains 
+the super block, which describes the amount and position (on the disk) of 
+metadata blocks. 
+
+2. **Mapping-table blocks**: After the super block, there is a set of blocks 
+that describes the mapping of (1) the logical chunks of the target logical 
+device to (2) data zones. The mapping is indexed by logical chunk number, and 
+each mapping entry indicates the data zone that stores the chunk data. 
+It can also indicate the zone number of a random zone that is used to 
+buffer random modifications to the chunk data.
+
+3. **Bitmap-storage blocks**: After the mapping-table blocks, there is a set of 
+blocks used to store bitmaps that indicate the validity of blocks in 
+the data zones. A valid block is any block that was written and not discarded. 
+In a buffered data zone, a block can be valid only (1) in the data zone or (2)
+in the buffer zone.
+
+To protect internal metadata against corruption in cases of sudden power loss 
+or system crashes, two sets of metadata zones are used. One set (the primary 
+set) is used as the main metadata set. The other set (the secondary set) is 
+used as a log. Modified metadata are first written to the secondary set and 
+the log that is created by that act of writing to the secondary set is 
+validated by writing an updated super block in the secondary set. After this 
+log operation completes, updates in place of metadata blocks can be done in the
```

Comment:
"updates in place of metadata blocks can be done" -- this whole thing is unclear, in my opinion.

Can you explain this to me simply?

Reply:
The primary metadata are that: primary, the "main" thing. The secondary set is a backup. The whole idea is to do micro-journaling by writing (updating) the secondary first, and if that is all OK, then write (update in-place) the primary.

Reply:
Ok. I get it now.

Let this sit in my head for a few hours while I figure out how to put this into plain(ish) English.

Reply:
I let this stew for 24 hours and read it again, and I found it necessary to make only a couple of small alterations.